File: src/ds-suite/libs/model/src/lib/basic-permissions.ts
// SPDX-FileCopyrightText: © 2014 Oberverwaltungsgericht Rheinland-Pfalz <poststelle@ovg.jm.rlp.de>
// SPDX-License-Identifier: EUPL-1.2
import { Restriction } from './restriction';

export class BasicPermissions {
    allowDisplays?: boolean = false;
    allowDisplaysControl?: boolean = false;
    allowTermine?: Restriction = Restriction.forbidden;
    allowNotes?: Restriction = Restriction.forbidden;

    public static isAdmin(perm: BasicPermissions) {
      return (perm.allowNotes == Restriction.write);
    }
  }

File: src/DigitalSignage.dn.WebApiCore/DtoModels/BasicPermissions.cs
// SPDX-FileCopyrightText: © 2014 Oberverwaltungsgericht Rheinland-Pfalz <poststelle@ovg.jm.rlp.de>
// SPDX-License-Identifier: EUPL-1.2
using DigitalSignage.WebApi.Controllers.Settings;

namespace DigitalSignage.dn.WebApiCore.DtoModels;

public class BasicPermissions
{
    public bool AllowDisplays { get; set; } = false;
    public bool AllowDisplaysControl { get; set; } = false;
    public Restriction AllowTermine { get; set; } = Restriction.forbidden;
    public Restriction AllowNotes { get; set; } = Restriction.forbidden;
}

public enum Restriction
{ 
    forbidden = 0, 
    read = 1, 
    write = 2 
};